### DISCUSSION TITLE
Versioning automation

### DISCUSSION THREAD

**Discussion by Karan-S-dev** [facebook/docusaurus#7927] [Q&A] [answered]
Hi All,

I created a docs site using docusaurus. In our site, we are having more that 100 products documentation. I  was able to do versioning using ,"npm run docusaurus docs:version 1.1.0".

But moving forward, I can't do manually versioning for all the products. So is there a way to automate versioning. Like without using npm, is there a way to do versioning?

If you need more clarity on the question also you can ask.

Thanks in advance.

**Comment by slorber** [comment, marked as answer]
I don't understand what you mean by "manual versioning". 

When you run `npm run docusaurus docs:version 1.1.0` is it really manual? Because Docusaurus does things for you 🤷‍♂️ 

If you need to version 100 products at once, you could as well run a script that runs `npm run docusaurus docs:version 1.1.0"` 100 times for each product? (I assume you use multiple docs plugins instances?)

Note the `docusaurus docs:version` CLI is just a convenience tool. You could easily do this all by yourself. What it does is just:
- copy docs to versioned_docs/version-1.1.0
- copy sidebars.js to versioned_sidebars/version-1.0.0-sidebars.json
- add "1.0.0" in versions.json

**Reply by Karan-S-dev** [reply]
Not manual, When i run npm command it does all the things automatically. Thats great but now we are trying to automate our site.
So in place of npm command, is there a way for me to do versioning or npm is the only preferred way?

**Reply by slorber** [reply]
Sorry, but I have no idea what you mean. What does it mean to "automate your site"? Why can't a script run npm commands exactly?
